source/platform/renderers/DirectX11/DirectX11.cpp

```diff
@@ -322,38 +321,41 @@ Font::Status DirectX11ResourceLoader::LoadFont(Font& font)
 
     pContext->Unmap(buftex, 0);
 
-    if (FAILED(m_pDevice->CreateShaderResourceView(buftex, nullptr, &data->m_Texture)))
+    if (FAILED(m_pDevice->CreateShaderResourceView(buftex, nullptr, &fontData->m_Texture)))
     {
-        delete data;
-        data = NULL;
+        delete fontData;
+        fontData = nullptr;
 
         GwkDxSafeRelease(buftex);
-
+        GwkDxSafeRelease(pContext);
         return font.status = Font::Status::ErrorBadData;
     }
 
     GwkDxSafeRelease(buftex);
+    GwkDxSafeRelease(pContext);
 
-    font.data = data;
+    font.data = fontData;
     font.status = Font::Status::Loaded;
 
-    m_FontList.push_back(&font);
+    m_FontList.push_back(&font); // Pointer to a reference, who owns this pointer?
 
     return font.status;
 }
 
 void DirectX11ResourceLoader::FreeFont(Font& font)
 {
+    // FontData* not deleted with Class' destruction
+
     if (font.IsLoaded())
     {
         if (font.data)
         {
             FontData* pFontData = static_cast<FontData*>(font.data);
             delete pFontData;
-            font.data = NULL;
+            font.data = nullptr;
         }
 
-        m_FontList.remove(&font);
+        m_FontList.remove(&font);  // Pointer to a reference, who owns this pointer?
```

Comment:
I think the Resource now owns it. TBH this is one of the reasons I forked GWEN, to fix some issue like this. I got a bit side tracked in other areas, but I still think the fonts needs some work, and you are right to point out the ownership issue. 

The ResourceLoaders are supposed to help here. The user could implement their own loader and still use this renderer, if they wanted to manage it themselves.